icaruscode/Decode/DataProducts/TriggerConfiguration.h

```diff
@@ -0,0 +1,268 @@
+#include <iosfwd> // std::ostream
+#include <string>
+#include <vector>
+
+namespace icarus {
+
+	struct TriggerConfiguration;
+
+  /// Prints the configuration into a stream with default verbosity.
+  std::ostream& operator<<(std::ostream& out, icarus::TriggerConfiguration const& config);
+
+}
+
+
+struct icarus::TriggerConfiguration {
+
+	// --- BEGIN -- Data members -------------------------------------------------
+  
+  // NOTE when adding data members, remember to add an element to the comparison
+
+  // Use the WR time reference
+  bool UseWrTime = true;
+
+  // Add an offset between the npt and tai time as used in the wr reference (normally it is 1 or 2 leap seconds)
+  unsigned int WrTimeOffset = 1e9;
+ 
+ 	// Veto (delay on the leading edge of the beam gate)
+ 	unsigned int VetoDelay = std::numeric_limits<unsigned int>::max();
+
+ 	// Majority level in-time cryo 1 (EAST)
+ 	unsigned int MajLevelBeamCryoEAST = std::numeric_limits<unsigned int>::max();
+
+ 	// Majority level out-of-time cryo 1 (EAST)
+ 	unsigned int MajLevelEnableCryoEAST = std::numeric_limits<unsigned int>::max();
+
+ 	// Sliding window option cryo 1 (EAST)
+ 	std::string SlidingWindowCryoEAST;
+
+ 	// Majority level in-time cryo 2 (WEST)
+ 	unsigned int MajLevelBeamCryoWEST = std::numeric_limits<unsigned int>::max();
+
+ 	// Majority level out-of-time cryo 2 (WEST)
+ 	unsigned int MajLevelEnableCryoWEST = std::numeric_limits<unsigned int>::max();
+
+ 	// Sliding window option cryo 2 (WEST)
+ 	std::string SlidingWindowCryoWEST;
+
+ 	// Majority trigger type ( consider trigger from one cryostats, either cryostats, or both cryostats )
+ 	std::string MajorityTriggerType;
+
+ 	// Run type ( MinBias: ignoring the light in-time or Majority: which applies a logic based on combination of distriminated light signals )
+ 	std::string RunType;
+
+ 	// TPCTriggerDelay: distance between the Global trigger time and the output for the TPC. NB: It is in units of 400 ns 
+ 	unsigned int TPCTriggerDelay = std::numeric_limits<unsigned int>::max();
+
+ 	// GateSelection: available gates to produce triggers: see registers 0x00000 in SBNDOCDB: 
+ 	std::string GateSelection ; 
+
+ 	// Duration of the conincidence gate synchronous with the BNB beam
+ 	unsigned int BNBBeamWidth = std::numeric_limits<unsigned int>::max();
+
+ 	// Duration of the drift window gate ( for out-of-time light activity ) synchronous with the BNB beam
+ 	unsigned int BNBEnableWidth = std::numeric_limits<unsigned int>::max();
+
+ 	// Duration of the conincidence gate synchronous with the NuMI beam
+ 	unsigned int NuMIBeamWidth = std::numeric_limits<unsigned int>::max();
+
+ 	// Duration of the drift window gate ( for out-of-time light activity ) synchronous with the NuMI beam
+ 	unsigned int NuMIEnableWidth = std::numeric_limits<unsigned int>::max();
+
+ 	// Ratio of spills to be collected ignoring the light coincidence ( MinBias ) compared to the total number of spills sent
+ 	std::string PreScaleBNBNuMI; 
```

Comment:
This is definitely something else than a string. I would use float or integer types.
Also, separating the BNB and NuMI is not a bad idea. Depending on how low level you want to go, you can even use a bit field:
```cpp
  struct {
    int BNB: 16;
    int NuMI: 16;
  } prescaleBeam;
```
(this is not the way _I_ would choose).